Makefile

```diff
@@ -30,14 +39,44 @@ e2e: deploy
 $(TARGETS): vendor/modules.txt
 	./scripts/$@
 
-build-subctl: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go")
+clean:
+	rm -f $(BINARIES) $(CROSS_BINARIES) $(CROSS_TARBALLS)
+
+build: operator-image $(BINARIES)
+
+build-cross: $(CROSS_TARBALLS)
+
+operator-image: vendor/modules.txt
+	operator-sdk build quay.io/submariner/submariner-operator:$(VERSION) --verbose
+
+bin/subctl: bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
+	ln -sf $(<F) $@
+
+dist/subctl-%.tar.xz: bin/subctl-%
+	mkdir -p dist
+	tar -cJf $@ --transform "s/^bin/subctl-$(VERSION)/" $<
+
+# Versions may include hyphens so it's easier to use $(VERSION) than to extract them from the target
+bin/subctl-%: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go") vendor/modules.txt
+	mkdir -p bin
+	target=$@; \
+	target=$${target%.exe}; \
+	components=($$(echo $${target//-/ })); \
+	GOOS=$${components[-2]}; \
+	GOARCH=$${components[-1]}; \
+	export GOARCH GOOS; \
+	$(SCRIPTS_DIR)/compile.sh \
+		--ldflags "-X github.com/submariner-io/submariner-operator/pkg/version.Version=$(VERSION) -X github.com/submariner-io/submariner-operator/pkg/version.GitRef=$(GIT_REF)" \
```

Comment:
Tangentially related, we can stop using GIT_REF since this was merged https://github.com/submariner-io/shipyard/pull/219

Reply:
Good point, https://github.com/submariner-io/submariner-operator/pull/507 fixes that (and if it’s merged before this, I’ll update this to suit, or vice versa as appropriate).